invokeai/app/api/dependencies.py

```diff
@@ -66,9 +72,25 @@ def initialize(config: InvokeAIAppConfig, event_handler_id: int, logger: Logger
         logger.debug(f"Internet connectivity is {config.internet_available}")
 
         output_folder = config.output_path
+        image_files = DiskImageFileStorage(f"{output_folder}/images")
 
         db = SqliteDatabase(config, logger)
 
+        migrator = SQLiteMigrator(
+            db_path=db.database if isinstance(db.database, Path) else None,
+            conn=db.conn,
+            lock=db.lock,
+            logger=logger,
+            log_sql=config.log_sql,
+        )
+        migration_2.register_post_callback(partial(migrate_embedded_workflows, logger=logger, image_files=image_files))
+        migrator.register_migration(migration_1)
+        migrator.register_migration(migration_2)
+        did_migrate = migrator.run_migrations()
+
+        if not db.is_memory and did_migrate:
+            db.reinitialize()
```

Comment:
We need to re-connect to the database after doing the migrations, but only if 1) there was a migration and 2) we are using a file database.

This closes the `SqliteDatabase`'s connection and runs its `__init__` logic again.

If we do this with a memory db, it wipes the db.

Reply:
This explanation was helpful. Can you add it as an in-code comment?

Reply:
Superseded